tic_tac_toe/src/board.py

```diff
@@ -0,0 +1,203 @@
+
+    def validate_move(self, x, y, player):
+        """
+        Takes in a move as 2-dim coordinates, and a player, and validates that it is a valid move
+        Parameters
+        ----------
+        x : integer or float
+        The first of the two 2-dim coordinates to be operated on
+
+        y : integer or float
+        The first of the two 2-dim coordinates to be operated on
+
+        player : string
+        "X" or "O", which player
+        Returns
+        ---------
+        : boolean
+        Whether or not it is a valid move
+        """
+        _index = self.convert(x, y)
+
+        if self.values[_index] == None:
+            return True
+
+        elif self.values[_index] == "X":
+            if player == "X":
+                return False
+
+            elif player == "O":
+                return True
+
+        elif self.values[_index] == "O":
+            if player == "O":
+                return False
+
+            elif player == "X":
+                return True
+
+    def check_column(self, board, x_or_o):
+        """
+        checks if a specific player has won with a column win
+        parameters
+        ----------
+        board : class
+        the board to operate on
+
+        x_or_o : string
+        which player we are checking for
+
+        returns
+        ----------
+        winner : boolean
+        whether or not that player won with a column win
+        """
+        board_width = 3
+        winner = None
+
+        for i in range(0, board_width):
+            values = set()
+            for j in range(0, board_width):
+                index = self.convert(i, j)
+                if board.values[index] != None:
+                    values.add(board.values[index])
+            if len(values) == 1:
+                if x_or_o == list(values)[0]:
+                    winner = True
+                    break
+                else:
+                    winner = False
+        return winner
+
+    def check_row(self, board, x_or_o):
+        """
+        Checks if a specific player has won with a row win
+        Parameters
+        ----------
+        board : class
+        The board to operate on
+
+        x_or_o : string
+        Which player we are checking for
+
+        Returns
+        ----------
+        winner : boolean
+        Whether or not that player won with a row win
+        """
+        board_width = 3
+        winner = None
+
+        for i in range(0, board_width):
+            values = set()
+            for j in range(0, board_width):
+                index = self.convert(i, j) + 1
+                if board.values[index] != None:
+                    values.add(board.values[index])
+            if len(values) == 1:
+                if x_or_o == list(values)[0]:
+                    winner = True
+                    break
+                else:
+                    winner = False
+        return winner
+
+    def check_diagonal(self, board, x_or_o):
```

Comment:
Reimplement this with a loop.
Also, this should be `def __check_diagonal(self)` and returns `X`, `O` or `None`